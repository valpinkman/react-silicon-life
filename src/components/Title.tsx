import * as React from 'react'

interface TitleProps {
  children: React.ReactNode;
}

const Title: React.FunctionComponent<TitleProps> = ({ children }: TitleProps) => <h1>{children}</h1>

export default Title
