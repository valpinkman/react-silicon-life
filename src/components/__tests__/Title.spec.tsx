import * as React from 'react'
import { render } from 'react-testing-library'

import Title from '../Title'

describe('<Title />', () => {
  test('render correctly', () => {
    const wrapper = render(<Title>Hello World</Title>)

    expect(wrapper).toMatchSnapshot()
  })

  test('renders with the right text', () => {
    const wrapper = render(<Title>Hello</Title>)

    expect(wrapper.container.textContent).toBe('Hello')
  })
})
