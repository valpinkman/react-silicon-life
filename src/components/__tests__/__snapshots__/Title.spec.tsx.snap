// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        Hello World
      </h1>
    </div>
  </body>,
  "container": <div>
    <h1>
      Hello World
    </h1>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
